adobe_vipm/adobe/mixins/order.py

```
@@ -537,24 +540,28 @@ def _process_new_lines(self, context: Context, flex_discounts: dict, payload: di
+        map_by_base_offer_subscriptions = {
+            get_partial_sku(offer_id): subs for offer_id, subs in offer_subscriptions.items()
+        }
 
         for line in context.upsize_lines:
             adobe_base_sku = line["item"]["externalIds"]["vendor"]
+            try:
+                adobe_subscription = map_by_base_offer_subscriptions[adobe_base_sku]
+            except KeyError:
+                raise ProcessingUpsizeLinesError(
+                    "Subscription has not been found in Adobe for sku %s.", adobe_base_sku
+                )
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)